functional_tests/functional_test.go

```diff
@@ -512,6 +531,69 @@ func testJavaTraces(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func testDotNetTraces(t *testing.T) {
+	tracesConsumer := setupOnce(t).tracesConsumer
+
+	var expectedTraces ptrace.Traces
+	expectedTracesFile := filepath.Join(testDir, expectedValuesDir, "expected_dotnet_traces.yaml")
+	expectedTraces, err := golden.ReadTraces(expectedTracesFile)
+	require.NoError(t, err)
+
+	waitForTraces(t, 30, tracesConsumer)
+	var selectedTrace *ptrace.Traces
+
+	require.Eventually(t, func() bool {
+		for i := len(tracesConsumer.AllTraces()) - 1; i > 0; i-- {
+			trace := tracesConsumer.AllTraces()[i]
+			if val, ok := trace.ResourceSpans().At(0).Resource().Attributes().Get("telemetry.sdk.language"); ok && strings.Contains(val.Str(), "dotnet") {
+				if expectedTraces.SpanCount() == trace.SpanCount() {
+					selectedTrace = &trace
+					break
+				}
+				selectedTrace = &trace
+				break
+			}
+		}
+		return selectedTrace != nil
+	}, 3*time.Minute, 5*time.Second)
+
+	require.NotNil(t, selectedTrace)
+
+	maskScopeVersion(*selectedTrace)
+	maskScopeVersion(expectedTraces)
+	maskSpanParentID(*selectedTrace)
+	maskSpanParentID(expectedTraces)
+
+	err = ptracetest.CompareTraces(expectedTraces, *selectedTrace,
+		ptracetest.IgnoreResourceAttributeValue("os.description"),
```

Comment:
Shouldn't this be covered on the golden file?

Reply:
The value can vary depending on the test runner host (local dev machine, Github workflow, remote Kubernetes cluster). I would consider trying to validate this value out of scope for now, we can improvement the value validation in the future.